Utils/Dataflow/pyDKB/common/_logging.py

```diff
@@ -0,0 +1,318 @@
+    def __init__(self, level, name='root'):
+        """ Initialize new root logger. """
+        Logger.__init__(self, name, level)
+
+
+class MultilineFormatter(logging.Formatter, object):
+    """ Formatter for multiline messages.
+
+    Every extra line (directly in the message body, or the traceback)
+    is:
+      * prefixed with '  (==)';
+      * suffixed with the part of format string which goes after
+                 '%(message)s' part.
+    """
+
+    _suffix = None
+
+    def __init__(self, *args):
+        """ Split format string into message format and suffix. """
+        new_args = list(args)
+        if len(args):
+            fmt = args[0]
+            new_args[0], self._suffix = self.splitFormat(fmt)
+        super(MultilineFormatter, self).__init__(*new_args)
+
+    def splitFormat(self, fmt):
+        """ Split format string into msg format and suffix.
+
+        Suffix is everything that goes after the message body.
+        """
+        format, suffix = ('', '')
+        splitted = fmt.split("%(message)s")
+        if len(splitted) > 1:
+            format = "%(message)s".join(splitted[:-1]) + "%(message)s"
+            suffix = splitted[-1]
+        else:
+            format = fmt
+        return format, suffix
+
+    def formatSuffix(self, record):
+        """ Return formatted suffix. """
+        return self._suffix % record.__dict__
+
+    def formatExtra(self, lines, prefix="  (==) ", suffix='', align=False):
+        """ Format extra lines of the log message (traceback, ...).
+
+        Parameter 'align' shows whether the suffix should be aligned
+        to the right (by the longest line), or to the left (as for normal
+        log messages).
+        """
+        if isinstance(lines, list) and len(lines):
+            max_len = len(max(lines, key=len))
+            # Need to add prefix len (in case that the longest line is
+            # among those that will be prefixed).
+            max_len += len(prefix)
+            if suffix and align:
+                line_fmt = "%%(line)-%ds" % max_len
+            else:
+                line_fmt = "%(line)s"
+            extra = line_fmt % {'line': lines[0]} + suffix
+            for line in lines[1:]:
+                extra += "\n" + line_fmt % {'line': prefix + line} + suffix
+        else:
+            extra = ""
+        return extra
+
+    def format(self, record):
+        """ Format multiline message.
+
+        Second and further lines from initial message are formatted
+        'extra' lines.
+        """
+        formatted = super(MultilineFormatter, self).format(record)
+        lines = formatted.splitlines()
+        suffix = self.formatSuffix(record)
+        result = self.formatExtra(lines, suffix=suffix, align=True)
+        return result
+
+
+# --------------------------
+# Module top-level functions
+# --------------------------
+
+
+def isInitialized():
+    """ Checks if the module (namely, root logger) is initialized. """
```

Comment:
Maybe "Check if ...", to conform with the other definitions and "Do X" template?